engine/src/flutter/lib/web_ui/lib/canvas.dart

```diff
@@ -50,6 +50,12 @@ abstract class PictureRecorder {
   factory PictureRecorder() => engine.renderer.createPictureRecorder();
   bool get isRecording;
   Picture endRecording();
+
+  /// Whether this reference to the underlying picture recorder is [dispose]d.
+  ///
+  /// This only returns a valid value if asserts are enabled, and must not be
+  /// used otherwise.
+  bool get debugDisposed;
```

Comment:
Is there a way to do this without changing the public API surface of `PictureRecorder`? Alternatively, we can add `debugDisposed` to the mobile version too.